### PR TITLE
Refactored warning message APIs

### LINKED ISSUE #614: Weird naming convention in SmackWarnings
To me it seems that functions `warnIfUnsound` and `warnUnsound` are also reporting incompleteness warnings.
For example, this `warning: over-approximating bitwise operation and` is an incompleteness warning if we are indeed over-approximating this operation. So I am confused about the naming convention here.

To me it seems we should split this into `warnUnsound` and `warnIncomplete` functions. Thoughts?

### PR DESCRIPTION
**Pull request by shaobo-he**
This commit changes incorrectly named `warn.*Unsound` into `warnIfIncomplete` or
`warnUnModeled` to better reflect their meanings. In the meanwhile, the
incorretly named warning level `unsound` gets changed to `imprecise`.

Fixes #614

### CODE REVIEW

File: lib/smack/SmackInstGenerator.cpp [hunk 1]
```diff
@@ -319,9 +319,9 @@ void SmackInstGenerator::visitInvokeInst(llvm::InvokeInst &ii) {
   if (f) {
     emit(rep->call(f, ii));
   } else {
-    // llvm_unreachable("Unexpected invoke instruction.");
-    SmackWarnings::warnUnsound("invoke instruction", currBlock, &ii,
-                               ii.getType()->isVoidTy());
+    llvm_unreachable("Unexpected invoke instruction.");
```
Comment (zvonimir) [inline lib/smack/SmackInstGenerator.cpp:322]: What is the reason for this change?

**Review by zvonimir — CHANGES_REQUESTED** [review]

File: lib/smack/SmackInstGenerator.cpp (hunk 1 above)
Comment (shaobo-he) [reply lib/smack/SmackInstGenerator.cpp:322]: I think the original handling (i.e., `llvm_unreachable`) makes more sense since we want to be seriously warned (i.e., by a crash) when an invoke instruction invokes a non-function object.

Comment (zvonimir) [reply lib/smack/SmackInstGenerator.cpp:322]: Ok, then please remove the comments. No need to leave them behind. We should also probably test this on SVCOMP one it gets merged.

Comment (shaobo-he) [reply lib/smack/SmackInstGenerator.cpp:322]: Done.

**Review by zvonimir — APPROVED** [review]
Looks good to me.
